concepts/0003-protocols/state-details.md

```diff
@@ -1,80 +1,39 @@
-# State Details and State Machines
+### State Details and State Machines
 
-Some protocols have only one sequence of states to manage. But in many
-protocols, different roles perceive the interaction differently. Each
-perspective needs to be described with care. This rigor has been neglected
-in many early efforts at protocol definition, and its omission is a big
-miss. Analyzing all possible states and events for all roles leads to
-robustness; skipping the analysis leads to fragility.
+While some protocols have only one sequence of states to manage, in most
+different roles perceive the interaction differently. The sequence of states
+for each role needs to be described with care in the RFC.
 
-### State machines
+#### State machines
 
-By convention, our community describes state and sequence rules using the
+By convention, protocol state and sequence rules are described using the
 concept of state machines, and we encourage developers who implement
 protocols to build them that way.
 
 Among other benefits, this helps with error handling: when one agent
-sends a [`problem-report`](../../features/0035-report-problem/README.md)
+sends a [`report-problem`](../../features/0035-report-problem/README.md)
 message to another, the message can make it crystal clear which state it
 has fallen back to as a result of the error.
 
-Many developers will have encountered a formal definition of this concept as
+Many developers will have encountered a formal of definition of state machines as
 they wrote parsers or worked on other highly demanding tasks, and may worry
 that state machines are heavy and intimidating. But as they are used in
-DIDComm protocols, state machines are straightforward and elegant. They
+Aries protocols, state machines are straightforward and elegant. They
 cleanly encapsulate logic that would otherwise be a bunch of conditionals
-scattered throughout agent code. Without a state machine:
-
-```python
-if received_message.type == X and state == Y:
-    do_something()
-...
-if received_message.type == X and state in [Y, Z]:
-    do_something_else()
-    state = V
-...
-if event == A and state in [V, W, Z]:
-    state = Y
-    do_yet_another_thing()
-``` 
-
-With a state machine:
-
-```python
-state_machine.handle(message) # hooks do_something()
-...
-state_machine.handle(message) # hooks do_something_else()
-...
-state_machine.handle(message) # hooks do_yet_another_thing()
-```
-
-Under maintenance, the robustness of the state machine approach is compelling.
-Compare the approaches for how many places you might have to change if you had
-a bug in how state Y is handled... The state machine approach also provides a
```

Comment:
Although the benefits of state machines as we use them are now tribal knowledge among some Aries implementers, I don't think it's a good idea to remove all the rationale here. It feels like  you're trying to make the RFC read faster. That's a noble goal. But for a concept RFC, I don't think that's the goal. Nor is simple conformance. Rather, the goal of a concept RFC is to guide a mental model. We don't accomplish that by suppressing all the discussion about why we arrived at the model we chose.

Reply:
My rationale in looking at this was that those interested in protocols would understand state-machines, but don't need to be sold on them. Those that want to see details/code can find that in the tictactoe example, which (I think) is more effective than the code snippets that were here.

Reply:
Okay, I can get on board with that.